src/util/glob.ts

```diff
@@ -212,20 +233,44 @@ export async function glob(
         braceExpansion: false,
       });
       for (const match of matches) {
-        combinedMap.set(match.path, match);
+        // Normalize the path so that:
+        //
+        // 1. We have native path separators. fast-glob returns "/" even on
+        //    Windows.
+        //
+        // 2. Trailing slashes and other remnants of the input patterns like
+        //    ".." are normalized away. fast-glob tends to preserve these in the
+        //    results. Note that `fs.resolve` trims trailing slashes, but
+        //    `fs.normalize` does not.
+        let path = pathlib.resolve(match.path);
+        if (
+          opts.throwIfOutsideCwd &&
+          !path.startsWith(normalizedCwdWithTrailingSep) &&
+          path !== normalizedCwd
```

Comment:
There's probably something I'm not thinking of but why do we need these 2 separate checks instead of just `!path.startsWith(normalizedCwd)`?